feat(oxlint): Generate markdownDescription fields for oxlint JSON schema.

connorshea opened the pull request:
This allows VS Code and other editors to display better descriptions in hover tooltips when editing oxlintrc.json files.

Before:

<img width="676" height="179" alt="Screenshot 2025-11-21 at 9 24 52 PM" src="https://github.com/user-attachments/assets/775cbf62-19cb-44f4-ad48-2a9e3b867900" />

After:

<img width="926" height="164" alt="Screenshot 2025-11-21 at 9 24 41 PM" src="https://github.com/user-attachments/assets/d2e668b2-88a9-4765-9f18-122185fdeb8e" />

This can probably be made more efficient, as I wrote it using GitHub Copilot + Claude Sonnet 4.5. But it's not really ever used outside developers regenerating the schema occasionally, so 🤷. Based heavily on https://github.com/moonrepo/schematic/blob/d3a9422cead2802b940c96f8ab8b188c86ab281f/crates/schematic/src/schema/renderers/json_schema.rs#L84, with permission (via discord).

Part of #15247.

camc314 commented:
### Merge activity

* **Nov 23, 12:22 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 23, 12:22 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
feat(oxlint): Generate markdownDescription fields for oxlint JSON schema. (#15959)

This allows VS Code and other editors to display better descriptions in hover tooltips when editing oxlintrc.json files.

Before:

<img width="676" height="179" alt="Screenshot 2025-11-21 at 9 24 52 PM" src="https://github.com/user-attachments/assets/775cbf62-19cb-44f4-ad48-2a9e3b867900" />

After:

<img width="926" height="164" alt="Screenshot 2025-11-21 at 9 24 41 PM" src="https://github.com/user-attachments/assets/d2e668b2-88a9-4765-9f18-122185fdeb8e" />

This can probably be made more efficient, as I wrote it using GitHub Copilot + Claude Sonnet 4.5. But it's not really ever used outside developers regenerating the schema occasionally, so 🤷. Based heavily on https://github.com/moonrepo/schematic/blob/d3a9422cead2802b940c96f8ab8b188c86ab281f/crates/schematic/src/schema/renderers/json_schema.rs#L84, with permission (via discord).

Part of #15247.
